libpromises/var_expressions.c

```diff
@@ -95,6 +96,13 @@ static size_t VarRefHash(const VarRef *ref)
     return (h & (INT_MAX - 1));
 }
 
+unsigned int VarRefHash_untyped(const void *ref,
+                                unsigned int seed ARG_UNUSED,
+                                unsigned int max)
+{
+    return VarRefHash(ref) & (max - 1);
+}
```

Comment:
@kacf performance is an issue, as well as memory usage since hash tables are created for each EvalContext now and 8192 is possibly too big. But I don't think this should block this pull request. It will leave in master anyway, and we ought to refactor our hash table (see [redmine#7966](https://dev.cfengine.com/issues/7966)).